apps/api/src/controllers/post.ts

```diff
@@ -0,0 +1,128 @@
+import Post from '../models/post';
+
+// Initialize posts array to save data in memory
+const posts = [];
+
+// Get all posts
+export const getPosts = async (req, res) => {
+  try {
+    // Return all the posts with a 200 status code
+    const posts = await Post.find();
+
+    res.status(200).json(posts);
+  } catch (error) {
+    const { message } = error;
+    res.status(500).json({ message });
+  }
+};
+
+const getPost = async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const post = await Post.findById(id);
+
+    if (!post) {
+      return res.status(404).json({ message: 'Post not found!' });
+    }
+    res.status(200).json(post);
+  } catch (error) {
+    const { message } = error;
+    res.status(500).json({ message });
+  }
+};
+
+// Get post by category
+const getPostsByCategory = async (req, res) => {
+  // Retrieve the id from the route params
+  const { category, id } = req.params;
+  try {
+    const postFound = await Post.findOne(category);
```

Comment:
Here we can use `Post.find({ category })` to get the posts by category, also you need to use `populate` method to get the information of the reference `Post.find().populate('category')`